modules/microshift-olm-deploy-ops-spec-ns.adoc

```diff
@@ -0,0 +1,185 @@
+
+[IMPORTANT]
+====
+All of the Operators installed in a specific namespace must have the same watch scope.
+====
+
+.Prerequisites
+* The {oc-first} is installed.
+* Operator Lifecycle Manager (OLM) is installed.
+* You have created a custom catalog that is running in the global namespace.
+
+.Procedure
+
+. Confirm that OLM is running by using the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-operator-lifecycle-manager get pod -l app=olm-operator
+----
++
+.Example output
+[source,terminal]
+----
+NAME                            READY   STATUS    RESTARTS   AGE
+olm-operator-66c97f894f-55wp2   1/1     Running   0          126m
+----
+
+. Confirm that the pod the catalog runs in is active by using the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-operator-lifecycle-manager get pod -l app=catalog-operator
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                READY   STATUS    RESTARTS   AGE
+catalog-operator-7cf77c459d-jbrd7   1/1     Running   0          126m
+----
+
+//Q which directory do we place these YAMLs in? do we need to specify?
+. Optional. Create a namespace by using the following command and example YAML:
++
+[source,terminal]
+----
+$ sudo cat _<ns.yaml>_ <1>
+----
+<1> Replace _<ns.yaml>_ with the name of your namespace configuration file.
++
+.Example namespace YAML
+[source,YAML]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: olm-microshift
+----
++
+. Apply the namespace configuration using the following command:
++
+----
+$ oc apply -f _<ns.yaml>_ <1>
+----
+<1> Replace _<ns.yaml>_ with the name of your namespace configuration file.
++
+.Example output
+[source,terminal]
+----
+namespace/olm-microshift created
+----
+
+. Create the Operator group YAML by using the following command and example YAML:
++
+[source,terminal]
+----
+$ sudo cat _<og.yaml>_ <1>
+----
+<1> Replace _<og.yaml>_ with the name of your operator group configuration file.
++
+.Example Operator group YAML
+[source,yaml]
+----
+kind: OperatorGroup
+apiVersion: operators.coreos.com/v1
+metadata:
+  name: og
+  namespace: olm-microshift
+spec: <1>
+  targetNamespaces:
+  - olm-microshift
+----
+<1> Omit the `spec.targetNamespaces` field and values for Operators using the global namespace.
+
+. Apply the Operator group configuration by running the following command:
++
+[source,terminal]
+----
+$ sudo oc apply -f _<og.yaml>_ <1>
+----
+<1> Replace _<og.yaml>_ with the name of your operator group configuration file.
++
+.Example output
+[source,terminal]
+----
+operatorgroup.operators.coreos.com/og created
+----
+
+. Create the `CatalogSource` object by using the following command and example YAML:
++
+[source,terminal]
+----
+$ sudo cat _<my-catalog-source.yaml>_ <1>
```

Comment:
I guess it should be `oc apply` instead of `sudo cat`?

Do we want to wait for readiness?

```
$ oc describe catalogsources.operators.coreos.com -n openshift-operators operatorhubio-catalog
Name:         operatorhubio-catalog
Namespace:    openshift-operators
Labels:       <none>
Annotations:  <none>
API Version:  operators.coreos.com/v1alpha1
Kind:         CatalogSource
Metadata:
  Creation Timestamp:  2024-01-30T12:05:14Z
  Generation:          1
  Resource Version:    59582
  UID:                 1260f123-1189-4b68-8153-841a56ee7a5d
Spec:
  Display Name:  Community Operators
  Grpc Pod Config:
    Security Context Config:  restricted
  Image:                      quay.io/operatorhubio/catalog:latest
  Publisher:                  OperatorHub.io
  Source Type:                grpc
  Update Strategy:
    Registry Poll:
      Interval:  60m
Status:
  Connection State:
    Address:                   operatorhubio-catalog.openshift-operators.svc:50051
    Last Connect:              2024-01-30T12:05:42Z
    Last Observed State:       READY
  Latest Image Registry Poll:  2024-01-30T13:05:14Z
  Registry Service:
    Created At:         2024-01-30T12:05:14Z
    Port:               50051
    Protocol:           grpc
    Service Name:       operatorhubio-catalog
    Service Namespace:  openshift-operators
Events:                 <none>
```

```
$ oc get pods -n openshift-operators -l olm.catalogSource=operatorhubio-catalog
NAME                          READY   STATUS    RESTARTS   AGE
operatorhubio-catalog-5tswm   1/1     Running   0          99m
```